pkg/pb/datarepair.proto

```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+syntax = "proto3";
+import "pointerdb.proto";
+package pb;
+
+// RepairSegment ..
+message RepairSegment {
```

Comment:
I am thinking of a little bit different queue item definition:
```go
message InjuredSegment {
    string path = 1;
    repeated int32 lost_pieces = 2;
}
```
The data repairer would want to know just these two thiings:
- which segment needs a repair
- which pieces got lost

All the rest of the info is available in the PointerDB and the data repairer could look it up from there. Having just the segment's path and the indices of the lost pieces is enough and would minimize the memory allocated by the queue.

I also started a Data Repairer Interfaces document: https://docs.google.com/document/d/17LXgm8vMI0PyZmZ9PpJSl1qDyDxPosQf38A7qEFoYxw/edit?usp=sharing

It should make it clear why the above info is enough.

Reply:
whoops! I'm now remembering alex asked kaloyan for this design doc in sprint planning, but also asked dennis for team rocket to do it earlier, and i completely spaced it in team rocket planning. sorry for the work overlap

kaloyan, jen had a good point here: https://storj.slack.com/archives/CA189C9DZ/p1537384363000100

perhaps which pieces are bad are different by the time the worker gets to it?

Reply:
I would imagine that the checker will update the queue as soon as new info about a segment is found. If the checker finds that a segment has lost more pieces and it is still in the queue, it won't add a new item to the queue, but it will update the existing one. And even move it more to the head of the queue if the risk for losing this segment is higher than other segments.

Similarly the checker may move an item to the tail if the risk for it has decreased, or even remove it completely from the queue if the risk for losing the segment is not real anymore.

This way when the repairer takes an item from the head, the info for the lost pieces will be as up-to-date as possible. The repairer won't have any responsibility to double-check the info from the repairer. It will just execute the repair as soon as possible.

Reply:
being able to edit existing queue items creates a significant increase in complexity, not to mention queue reordering! instead, we could just expect that every day (for the foreseeable future), the queue gets all at risk items added again, and we should work to drain the queue every day.

how are you expecting to implement this queue with editing and prioritization features?

Reply:
So do we want to just list the pieces, or pass in the whole segment still?